rfd/0203-database-healthchecks.md

```diff
@@ -0,0 +1,606 @@
+
+Related issues:
+- [Database Access upstream DB health checks](https://github.com/gravitational/teleport/issues/20544)
+- [Desktop Access: support latency-based routing to a desktop service](https://github.com/gravitational/teleport/issues/40905)
+- [Detect and report latency measurements for Desktop sessions in the UI](https://github.com/gravitational/teleport/issues/35691)
+
+## Details
+
+### UX
+
+The web UI will be updated to leverage database health status information.
+
+Unhealthy databases will be displayed with a warning icon or tooltip icon indicating that the database is unhealthy.
+
+Recall that there may be multiple `db_server` heartbeats for the same database, but the web UI only shows a single resource tile for the database.
+If this is the case, then the tooltip will be shown if any of the heartbeats have an unhealthy `target_health.status`.
+The tooltip content may look something like this (or equivalent singular form verbiage if there's only one agent):
+
+    M out of N Teleport database services proxying access to this database cannot reach the database endpoint.
+    <link to healthcheck/network troubleshooting doc>
+
+    Affected Teleport database services:
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Error: <last health check error message>
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Error: <last health check error message>
+
+Databases with a mix of `healthy` and `""` (unknown) status will also show a tooltip that encourages the user to enable health checks or update to a supported version on all of the agents that proxy the db.
+The tooltip content may look something like this: 
+
+    M out of N Teleport database services proxying access to this database are not running network health checks for the database endpoint.
+    User connections will not be routed through affected Teleport database services as long as other database services report a healthy connection to the database.
+
+    Affected Teleport database services:
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Reason: The database service version v17.1.2 does not support health checks.
+    - Hostname: <hostname>
+      UUID: <uuid>
+      Reason: The database service has disabled health checks for this database.
+
+#### User story: AWS RDS database enrollment
+
+A Teleport cluster admin is new to Teleport and would like to enroll an AWS RDS database with Teleport.
+
+- The user logs into Teleport, sees the "Enroll New Resource" button and clicks that.
+- They click on the "RDS PostgreSQL" tile - a guided database enrollment wizard.
+- After selecting an AWS integration, region, and VPC, the user sees the "Enroll RDS Database" page.
+- On the page is a toggle button labeled "Monitor database endpoint health", which is already enabled by default.
+- The user disables the toggle, which displays a warning, so they toggle it back to the default enabled state.
```

Comment:
Why do we need a toggle? Let's just have it always enabled by default. The fewer toggles we have the better and I don't see a reason anyone would not want to monitor their database health. I think in this case it's ok to be opinionated and it would result in better UX as well.